main.go

```diff
@@ -161,65 +167,118 @@ func NewNSMClient(ctx context.Context, rootConf *config.Config) networkservice.N
 		spiffejwt.TokenGeneratorFunc(source, rootConf.MaxTokenLifetime),
 		clientCC,
 		sendfd.NewClient(),
+		token.NewClient(),
 	)
 }
 
+// Connection is a return type for RunClient
+type Connection struct {
+	Client     networkservice.NetworkServiceClient
+	Connection *networkservice.Connection
+	Cancel     context.CancelFunc
+}
+
 // RunClient - runs a client application with passed configuration over a client to Network Service Manager
-func RunClient(ctx context.Context, rootConf *config.Config, nsmClient networkservice.NetworkServiceClient) ([]*networkservice.Connection, error) {
+func RunClient(ctx context.Context, rootConf *config.Config, nsmClient networkservice.NetworkServiceClient) ([]*Connection, error) {
```

Comment:
Probably Connection struct can be removed from main.go if we change the signrature 
```suggestion
func RunClient(ctx context.Context, rootConf *config.Config, nsmClient networkservice.NetworkServiceClient) ([]*networkservice.Connection, context.CancelFunc, error)
```
Where CanceFunc should close all clients.

Reply:
Reworked with:
```go
func RunClient(...)(cleanup func(), err error) {...}
```
and so `cleanup()` closes all created connections.